File: cpp/problems/leetcode/0035_search_insert_position/solution.cpp
// LeetCode 35
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        return (int)(lower_bound(nums.begin(), nums.end(), target) - nums.begin());
    }
};

int main() {
    vector<int> a{1, 3, 5, 6};
    assert(Solution().searchInsert(a, 5) == 2);
    assert(Solution().searchInsert(a, 2) == 1);
    assert(Solution().searchInsert(a, 7) == 4);
    assert(Solution().searchInsert(a, 0) == 0);
    cout << "0035 OK" << endl;
    return 0;
}

File: python/problems/leetcode/0035_search_insert_position/solution.py
"""LeetCode 35: Search Insert Position."""

from __future__ import annotations

from bisect import bisect_left


class Solution:
    def searchInsert(self, nums: list[int], target: int) -> int:
        return bisect_left(nums, target)


if __name__ == "__main__":
    assert Solution().searchInsert([1, 3, 5, 6], 5) == 2
    assert Solution().searchInsert([1, 3, 5, 6], 2) == 1
    assert Solution().searchInsert([1, 3, 5, 6], 7) == 4
    assert Solution().searchInsert([1, 3, 5, 6], 0) == 0
    print("0035 OK")
